src/user/library/index.ts

```diff
@@ -0,0 +1,209 @@
+async function createNotification(accessToken: string, createInput: CreativeInput) {
+  const response = await axios.post(
+    `${process.env.REACT_APP_SERVER_ADDRESS}`.replace("v1", "graphql"),
+    JSON.stringify({query: print(CreateCreativeDocument), variables: { input: createInput }}),
+    {
+      headers: {
+        "Authorization": `Bearer ${accessToken}`,
+        "Content-Type": "application/json"
+      }
+    }
+  );
+
+  return response.data.data.createCreative.id;
+}
+
+// TODO: Get rid of this ASAP. Currently necessary because when updating a campaign, it does not take into account any new ads.
+async function createAd(accessToken: string, createInput: CreateAdInput) {
+  const response = await axios.post(
+    `${process.env.REACT_APP_SERVER_ADDRESS}`.replace("v1", "graphql"),
+    JSON.stringify({query: print(CreateAdDocument), variables: { createAdInput: createInput }}),
+    {
+      headers: {
+        "Authorization": `Bearer ${accessToken}`,
+        "Content-Type": "application/json"
+      }
+    }
+  );
+
+  return response.data.data.createAd.id;
+}
+
+export function editCampaignValues(campaign: CampaignFragment): CampaignForm {
+  return {
+    adSets: campaign.adSets.map((adSet) => ({
+      ...adSet,
+      id: adSet.id,
+      conversions: adSet.conversions ?? [],
+      oses: adSet.oses ?? [] as OS[],
+      segments: adSet.segments ?? [] as Segment[],
+      billingType: adSet.billingType ?? "cpm",
+      name: adSet.name || adSet.id.split("-")[0],
+      creatives: adSet.ads!.map((ad) => {
+        const c = ad.creative;
+        return {
+          id: c.id,
+          name: c.name,
+          targetUrl: c.payloadNotification!.targetUrl,
+          title: c.payloadNotification!.title,
+          body: c.payloadNotification!.body
+        }
+      }),
+      price: adSet.ads?.[0].prices[0].amount || 0
+    })),
+    budget: campaign.budget,
+    currency: campaign.currency,
+    dailyBudget: campaign.dailyBudget,
+    dailyCap: campaign.dailyCap,
+    endAt: campaign.endAt,
+    format: campaign.format,
+    geoTargets: campaign.geoTargets ?? [] as GeocodeInput[],
+    name: campaign.name,
+    pacingStrategy: campaign.pacingStrategy,
+    startAt: campaign.startAt,
+    state: campaign.state,
+    type: "paid"
+  };
+}
+
+export async function transformEditForm(
+  form: CampaignForm,
+  id: string,
+  auth: any,
+  advertiserId: string
+): Promise<UpdateCampaignInput> {
+  const transformedAdSet: UpdateAdSetInput[] = [];
+
+  for (let adSet of form.adSets) {
+    const newCreatives = adSet.creatives.filter((c) => c.id === undefined);
+    for (let ad of newCreatives) {
+      const withId = await transformCreative(
+        ad,
+        adSet,
+        auth.accessToken,
+        advertiserId
+      );
+      await createAd(auth.accessToken, {
+        ...withId,
+        creativeSetId: adSet.id,
+      });
+    }
```

Comment:
Only create a creative when no id exists, then create the ad relation for it. Needed to be done for reasons specified above.